src/Time/Units.hs

```diff
@@ -285,3 +287,23 @@ getCPUTime :: forall unit m .
               (KnownRat unit, KnownRat (PicoSecondUnit / unit), MonadIO m)
            => m (Time unit)
 getCPUTime = liftIO CPUTime.getCPUTime >>= pure . convertUnit . ps . fromInteger
+
+{- | Similar to 'Timeout.timeout' but receiving any time unit
+instead of number of microseconds.
+
+>>> timeout (sec 1) (putStrLn "Hello O'Clock")
+Hello O'Clock
+Just ()
+
+>>> timeout (ps 1) (putStrLn "Hello O'Clock")
+Nothing
+
+>>> timeout (mcs 1) (putStrLn "Hello O'Clock")
+HellNothing
+
+-}
+timeout :: forall unit m a . (MonadIO m, KnownRat unit, KnownRat (unit / MicroSecondUnit))
+        => Time unit   -- ^ time
+        -> IO a        -- ^ 'IO' action
+        -> m (Maybe a) -- ^ returns 'Nothing' if no result is available within the given time
+timeout t = liftIO . Timeout.timeout (floor $ convertUnit @MicroSecondUnit t )
```

Comment:
This can be written in point-free style like:

```haskell
timeout = liftIO . Timeout.timeout . floor . convertUnit @MicroSecondUnit
```

With `-XTypeApplications` you can partially apply types as well.

Reply:
Unfortunately, it can't be done so easily because there is another argument which is eta-reduced now

Reply:
Ahh, sorry, my bad :slightly_frowning_face: